src/App.tsx

```diff
@@ -15,9 +15,23 @@ import { SingleJob } from "./pages/SingleJob/SingleJob";
 const App: React.FC<{}> = (): React.ReactElement => {
 	const [loggedIn, setLoggedIn] = React.useState(false);
 	const [openModal, setOpenModal] = React.useState(false);
+	const [showVacancy, setShowVacancy] = React.useState(false);
+	const [jobForm, setJobForm] = React.useState({
+		jobTitle: '',
+		companyName: '',
+		location: '',
+		companyDescription: '',
+		salaryMin: '',
+		salaryMax: '',
+		endDate: '',
+		jobDescription: '',
+		essentialRequirments: [''],
```

Comment:
i think default state here can just be an empty array, I think this will give it an extra element of empty string

Reply:
If I remove the empty string from the array I get  the error 

Type 'Dispatch<SetStateAction<{ jobTitle: string; companyName: string; location: string; companyDescription: string; salaryMin: string; salaryMax: string; endDate: string; jobDescription: string; essentialRequirments: never[]; desiredRequirments: never[]; responsibilities: never[]; }>>' is not assignable to type '(jobForm: { jobTitle: string; companyName: string; location: string; companyDescription: string; salaryMin: string; salaryMax: string; endDate: string; jobDescription: string; essentialRequirments: string[]; desiredRequirments: string[]; responsibilities: string[]; }) => void'.
  Types of parameters 'value' and 'jobForm' are incompatible.
    Type '{ jobTitle: string; companyName: string; location: string; companyDescription: string; salaryMin: string; salaryMax: string; endDate: string; jobDescription: string; essentialRequirments: string[]; desiredRequirments: string[]; responsibilities: string[]; }' is not assignable to type 'SetStateAction<{ jobTitle: string; companyName: string; location: string; companyDescription: string; salaryMin: string; salaryMax: string; endDate: string; jobDescription: string; essentialRequirments: never[]; desiredRequirments: never[]; responsibilities: never[]; }>'.
      Type '{ jobTitle: string; companyName: string; location: string; companyDescription: string; salaryMin: string; salaryMax: string; endDate: string; jobDescription: string; essentialRequirments: string[]; desiredRequirments: string[]; responsibilities: string[]; }' is not assignable to type '{ jobTitle: string; companyName: string; location: string; companyDescription: string; salaryMin: string; salaryMax: string; endDate: string; jobDescription: string; essentialRequirments: never[]; desiredRequirments: never[]; responsibilities: never[]; }'.
        Types of property 'essentialRequirments' are incompatible.
          Type 'string[]' is not assignable to type 'never[]'.
            Type 'string' is not assignable to type 'never'.